File: Backend/Models/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Backend.Models
{
    public class Product
    {
        public int ProductID { get; set; }
        public string Name { get; set; }
        public int? ArtistID { get; set; }
        public int? FormatID { get; set; }
        public decimal Price { get; set; }
        public int Stock { get; set; }
        public string Description { get; set; }

        public Artist Artist { get; set; }
        public Format Format { get; set; }
        public ICollection<ProductGenre> ProductGenres { get; set; }
        public ICollection<Image> Images { get; set;}
    }
}

File: Backend/src/models/product.ts
import mongoose from "mongoose";
import { ProductType } from "../shared/types";

const productSchema = new mongoose.Schema({
  name: { type: String, required: true },
  genre: [
    {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Genre",
      required: true,
    },
  ],
  artist: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "Artist",
    required: true,
  },
  format: { type: String, required: true },
  releaseDate: { type: Date, required: true },
  price: { type: Number, required: true },
  description: { type: String, required: true },
  stock: { type: Number, required: true },
  imageUrls: [{ type: String, required: true }],
  lastUpdated: { type: Date, required: true },
});

const Product = mongoose.model<ProductType>("Product", productSchema);
export default Product;
